regulation-worker/internal/delete/batch/batch.go

```diff
@@ -144,115 +113,106 @@ func (*Batch) updateStatusTrackerFile(absStatusTrackerFileName, fileName string)
-func (b *Batch) decompress(compressedFileName string) (string, error) {
-	compressedFilePtr, err := os.OpenFile(compressedFileName, os.O_RDWR, 0o644)
-	if err != nil {
-		return "", fmt.Errorf("error while opening compressed file, %w", err)
-	}
-	defer compressedFilePtr.Close()
+func (b *Batch) cleanedFiles(_ context.Context, f io.ReadWriteCloser, job *model.Job) ([]string, error) {
+	pkgLogger.Debugf("fetching already cleaned files based on the status tracker file")
 
-	gzipReader, err := gzip.NewReader(compressedFilePtr)
-	if err != nil {
-		return "", fmt.Errorf("error while reading compressed file: %w", err)
-	}
+	var response []string
+	defer f.Close()
 
-	decompressedFilePtr, err := os.CreateTemp(b.TmpDirPath, "")
+	byt, err := io.ReadAll(f)
 	if err != nil {
-		return "", fmt.Errorf("error while creating temporary file for decompressed files during cleaning")
+		return nil, fmt.Errorf("unable to read contents of status tracker file, err: %w", err)
 	}
-	defer decompressedFilePtr.Close()
+	// if statusTracker.txt exists then read it & remove all those files name from above gzFilesObjects,
+	// since those files are already cleaned.
 
-	_, err = io.Copy(decompressedFilePtr, gzipReader)
-	if err != nil {
-		return "", fmt.Errorf("error while writing uncompressed file: %w", err)
-	}
+	jobID := fmt.Sprintf("%d", job.ID)
 
-	decompressedFileName, err := filepath.Abs(decompressedFilePtr.Name())
-	if err != nil {
-		return "", err
-	}
+	if len(byt) == 0 {
+		// insert <jobID> in 1st line
+		if _, err := io.WriteString(f, jobID+"\n"); err != nil {
+			return nil, fmt.Errorf("error while writing to jobId to statusTrackerFile: %w", err)
+		}
+	} else {
+		lines := strings.Split(string(byt), "\n")
+		// check if our <jobID> matches with the one in file.
+		// if not, then truncate the file & write new current jobID.
 
-	return decompressedFileName, nil
-}
+		// TODO: Why would this happen ? ( remanent of the past )
+		if lines[0] != jobID {
+			f.Close()
 
-// compress & write `cleanedBytes` of type []byte to `fileName`
-func (*Batch) compress(fileName string, cleanedBytes []byte) error {
-	// compressing
-	var buffer bytes.Buffer
-	w := gzip.NewWriter(&buffer)
-	_, err := w.Write(cleanedBytes)
-	if err != nil {
-		return fmt.Errorf("error while compressing file: %w", err)
-	}
-	w.Close() // must close this first to flush the bytes to the buffer.
+			statusTrackerTmpDir, err := os.MkdirTemp(b.TmpDirPath, "")
+			if err != nil {
+				return nil, fmt.Errorf("error while creating temporary directory: %w", err)
+			}
 
-	cleanCompressedFilePtr, err := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
-	if err != nil {
-		return fmt.Errorf("error while opening file, %w", err)
-	}
-	defer cleanCompressedFilePtr.Close()
+			f, err = os.OpenFile(filepath.Join(statusTrackerTmpDir, StatusTrackerFileName), os.O_CREATE|os.O_RDWR, 0o644)
+			if err != nil {
+				return nil, fmt.Errorf("error while opening file to create status tracker, %w", err)
```

Comment:
[minor] more idiomatic error messages
```suggestion
				return nil, fmt.Errorf("create status tracker: %w", err)
```

Reply:
Old habits 😄